library/alloc/src/collections/vec_deque/mod.rs

```diff
@@ -154,11 +133,13 @@ unsafe impl<#[may_dangle] T, A: Allocator> Drop for VecDeque<T, A> {
             }
         }
 
-        let (front, back) = self.as_mut_slices();
-        unsafe {
-            let _back_dropper = Dropper(back);
-            // use drop for [T]
-            ptr::drop_in_place(front);
+        if mem::needs_drop::<T>() {
```

Comment:
nit: the convention in the library is not to have this to avoid *O(1)* work -- the `ptr::drop_in_place` call already knows to do nothing if the type doesn't need dropping, so we don't duplicate it here.

(if you needed a `.for_each(drop)` *loop*, then guarding it with `needs_drop` would be correct, but to just call `drop_in_place` twice the check isn't needed.)